### PR TITLE
work in progress for refactoring formatting of days in the header

### PR DESCRIPTION
**Pull request by z3cka**
@akohler not done yet, but you get the idea... any thoughts?

### CODE REVIEW

File: www/sites/all/modules/custom/ucla_libhours/ucla_libhours.module [hunk 1]
```diff
@@ -319,6 +305,34 @@ function ucla_libhours_get_library_hours($endpoint, $period, $library, $length)
 }
 
 /**
+ * Utility function that returns a week range as a string like: "May 21 - May 27"
+ */
+function ucla_libhours_format_week_range($start_date, $end_date) {
```
Comment (akohler) [inline www/sites/all/modules/custom/ucla_libhours/ucla_libhours.module]: Technically a date range, not week - nothing constraining it to 7 days or Sun-Sat.  Nitpicky, naming is hard, etc.

File: www/sites/all/modules/custom/ucla_libhours/ucla_libhours.module [hunk 2]
```diff
@@ -319,6 +305,34 @@ function ucla_libhours_get_library_hours($endpoint, $period, $library, $length)
...
+  return format_date(strtotime($start_date), 'custom', "M j") ." - ". format_date(strtotime($end_date), 'custom', "M j");;
```
Comment (akohler) [inline www/sites/all/modules/custom/ucla_libhours/ucla_libhours.module]: I would pull `format_date(strtotime($date), 'custom', "M j")` into a separate function like `ucla_libhours_format_as_monthday($date)` or similar.  Repeated code, unclear function; your initial comment documents its use as-is, but doesn't support re-use.  Same with the ones which format just day of month - utility function.  You could cheat and pass the format string too, though that muddies the purpose of the utility function.

Did I mention: big fan of Fowler's Refactoring book?

Comment (akohler) [inline www/sites/all/modules/custom/ucla_libhours/ucla_libhours.module]: Also, minor typo with two semicolons at end of line 311

File: www/sites/all/modules/custom/ucla_libhours/ucla_libhours.module [hunk 3]
```diff
@@ -25,14 +25,15 @@ function ucla_libhours_get_hours($library, $length) {
   // $latest_week = $length - 1;
 
   // put header day values in a keyed array
+  // TODO: use ucla_libhours_format_week_header_days() in a nested fashion
   foreach($element['hours']['locations'][0]['weeks'] as $week => $days) {
     foreach ($element['hours']['locations'][0]['weeks'][$week] as $day => $value) {
       $element['hours']['locations'][0]['weeks'][$week]['days'][$day] = $element['hours']['locations'][0]['weeks'][$week][$day];
```
Comment (akohler) [inline www/sites/all/modules/custom/ucla_libhours/ucla_libhours.module]: Here and many other places: does it make sense to pull a reference to `$element['hours']['locations'][0]['weeks']` (for example) into its own variable?  It adds a line or 2 but could make things much more readable, with these really deep arrays.  Something like (typed manually, untested...):
```
$weeks = $element['hours']['locations'][0]['weeks'];
foreach($weeks as $week => $days) {
  foreach($weeks[$week] as $day => $value) {
    $weeks[$week]['days'][$day] = $weeks[$week][$day];
    // etc
  }
}
```

File: www/sites/all/modules/custom/ucla_libhours/ucla_libhours.module [hunk 4]
```diff
@@ -319,6 +305,34 @@ function ucla_libhours_get_library_hours($endpoint, $period, $library, $length)
...
+
+/**
+ * Utility function that adds formatted day values to arrays
+ */
+function ucla_libhours_format_week_header_days(&$days, &$locations) {
+  foreach($days as $day => $date) {
+    $days[$day]['abrv'] = strtolower(substr($day, 0, 3));
+    // for Sat, Sun & Tues, Thurs use 2 characters else use 1
+    if (substr($day, 0, 1) === 'S' || substr($day, 0, 1) === 'T') {
+      $days[$day]['label'] = substr($day, 0, 2) ." ". format_date(strtotime($locations[0]['weeks'][0][$day]['date']), 'custom', "j");
```
Comment (akohler) [inline www/sites/all/modules/custom/ucla_libhours/ucla_libhours.module]: Same as above with "M j" date formatting.

**Comment by z3cka** [comment]
I still have a refactor to to finish regarding calling [`TODO: use ucla_libhours_format_week_header_days() in a nested fashion`](https://github.com/z3cka/libraryweb-site/pull/22/files#diff-98680f21db713cccd6b4f06f7b0181ffR28) but I'm going to merge this at its current state as we are deploying to `www-test` for QA today.

**Pull request merged**